signatures.md

```diff
@@ -52,163 +49,359 @@ Note: `Message` is used here as the object being signed, but this interface shou
 ```go
 type Signature interface {
 
-        // Sign generates a proof that miner `M` generate message `m`
+        // Sign generates a proof that miner `M` generated or approved of message `msg`.
         //
         // Out:
-        //    sig - a series of bytes representing a signature usually `r`|`s`
-        //    err - a standard error message indicating any process issues
+        //   sig - a series of bytes representing a signature  
+        //   err - a standard error message indicating any process issues
         // In:
-        //    m - a series of bytes representing a message to be signed
-        //    sk - a private key which cryptographically links `M` to `sig`
+        //   msg - a series of bytes representing a message to be signed
+  		  //   addr - the Filecoin `Address` of the signer
+        //   sk - a private key which cryptographically links `M` to `sig`
         //
-        Sign(m Message, sk PrivateKey) (sig SignatureBytes, err error)
-
-        // Verify validates the statement: only `M` could have generated `sig`
+        Sign(msg Message, addr Address, sk PrivateKey) (sig SignatureBytes, err error)
```

Comment:
this feels kinda weird. the sign method is on the signature itself. So i need to have a signature to sign something? seems like the sign method should be on a private key, or some other 'signaturescheme' or 'privkeymanager' type thing (those names are probably bad, but i'm sure theres something better here)

Reply:
Agree. Definitely feels weird. This should be captured in a `privateKey` method for BLS signatures. I don't believe we have a specification for private keys. Maybe this should be in data-structures.md? Context: If I recall correctly, I put this method here (last year) as a placeholder so we captured the`Sign` function somewhere.

Alternatively, we could place the `privateKey` specification here instead of data-structures.md. Open to suggestions.

(FYI: @dignifiedquire)

Reply:
@frrist: As one of the core devs, do you have a strong preference on where this new method should be documented?

Reply:
I think documenting this in the spec is fine for now. Once we implement it the documentation can be added to the implementation as well.

Reply:
Added an interface for `PrivateKey`. This will require further work. Please review.

Reply:
Adding a `PrivateKey` interface sounds reasonable to me.